### PR TITLE
Fix unexpected websocket handshake failure degrading into infinite loop

### PR DESCRIPTION
**Pull request by CandyAngel**
### References
This PR has an [alternative PR](https://github.com/kraih/mojo/pull/1018) which implements the same fix in a different way (less risk of side effects, clearer code [in my opinion]).

### CODE REVIEW

File: t/mojo/websocket_proxy.t [hunk 1]
```diff
@@ -199,6 +199,7 @@ my ($success, $leak, $err);
 $ua->websocket(
   "ws://127.0.0.1:$dummy/test" => sub {
     my ($ua, $tx) = @_;
+    Mojo::IOLoop->stop and return unless $tx->is_websocket;
```
Comment (kraih) [inline t/mojo/websocket_proxy.t]: The return value of `Mojo::IOLoop->stop` is undefined.

**Comment by kraih** [comment]
Same as #1018, i have to vote 👎.

**Comment by kraih** [comment]
Maybe worth mentioning too, i find the fact that there are two competing pull requests very confusing, and the descriptions don't make a clear case for why either one should be picked.

**Comment by kraih** [comment]
If you are not confident enough to pick one solution, perhaps it was too early to send a pull request.

**Comment by CandyAngel** [comment]
Closing this as I think #1018 is the better implementation.

**Pull request closed without merging**